Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,459 @@
+    }
+
+    $newGitHubRepositoryParms = @{
+        AutoInit = $true
+        Private = $true
+        RepositoryName = $defaultRepositoryName
+        OrganizationName = $script:organizationName
+    }
+    $repo = New-GitHubRepository @newGitHubRepositoryParms
+    # Get the main/master branch name
+    $mainBranchName = $repo | Get-GitHubRepositoryBranch | Select-Object -ExpandProperty name -First 1
+}
+
+Describe 'GitHubCodespaces\Delete-GitHubCodespace' {
+    Context 'When deleting a codespace for the authenticated user' {
+        BeforeEach {
+            # Suppress HTTP 202 warning for codespace creation
+            # TODO: Suppression is not working as intended here.
+            $WarningPreference = 'SilentlyContinue'
+
+            $newGitHubCodespaceParms = @{
+                OwnerName = $script:organizationName
+                RepositoryName = $defaultRepositoryName
+                Wait = $true
+            }
+            $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+        }
+
+        It 'Should get no content using -Confirm:$false' {
+            # Also asserts pipeline input
+            $codespace | Remove-GitHubCodespace -Confirm:$false
+            { Get-GitHubCodespace -CodespaceName $codespace.name } | Should -Throw
+        }
+
+        It 'Should get no content using -Force' {
+            # Also assert CodespaceName input
+            Remove-GitHubCodespace -CodespaceName $codespace.name -Force
+            { Get-GitHubCodespace -CodespaceName $codespace.name } | Should -Throw
+        }
+    }
+}
+
+Describe 'GitHubCodespaces\Get-GitHubCodespace' {
+    BeforeAll {
+        # Suppress HTTP 202 warning for codespace creation
+        $WarningPreference = 'SilentlyContinue'
+
+        $newGitHubCodespaceParms = @{
+            OwnerName = $script:organizationName
+            RepositoryName = $defaultRepositoryName
+            Wait = $true
+        }
+        $null = New-GitHubCodespace @newGitHubCodespaceParms
+    }
+
+    Context 'When getting codespaces for the authenticated user' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace |
+                Where-Object { $_.repository.name -eq $defaultRepositoryName }
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespaces[0].PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return one or more results' {
+            $codespaces.Count | Should -BeGreaterOrEqual 1
+        }
+
+        It 'Should return the correct properties' {
+            foreach ($codespace in $codespaces)
+            {
+                $codespace.owner.login | Should -Be $script:OwnerName
+                $codespace.repository.name | Should -Be $newGitHubCodespaceParms.RepositoryName
+            }
+        }
+    }
+
+    Context 'When getting a codespace for a specified owner and repository' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace @newGitHubCodespaceParms
```

Comment:
`Get-GitHubCodespace` doesn't have `Wait`, but `$newGitHubCodespaceParms` does.  That results in this failure:

``
[-] Context GitHubCodespaces\Get-GitHubCodespace.When getting a codespace for a specified owner and repository failed
 ParameterBindingException: A parameter cannot be found that matches parameter name 'Wait'. This failure might be caused by applying the default parameter binding. You can disable the default parameter binding in $PSDefaultParameterValues by setting $PSDefaultParameterValues["Disabled"] to be $true, and retry. The following default parameter was successfully bound for this cmdlet when the error occurred: -OutVariable
 at <ScriptBlock>, D:\Source\PowerShellForGitHub\tests\GitHubCodespaces.tests.ps1:104
``